### ISSUE TITLE
Error in julia 0.5

### ISSUE THREAD

**Issue by bdemeshev** [DynareTeam/Dynare.jl#2]
Pkg.add("Dynare")
Pkg.checkout("Dynare")
using Dynare
```julia
ERROR: LoadError: MethodError: no method matching Union(::Type{Expr}, ::Type{Symbol}, ::Type{Number})
Closest candidates are:
  Union{T}(::Any) at sysimg.jl:53
 in include_from_node1(::String) at ./loading.jl:488
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in eval(::Module, ::Any) at ./boot.jl:234
 in eval(::Module, ::Any) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in require(::Symbol) at ./loading.jl:415
 in require(::Symbol) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
while loading /Users/boris/.julia/v0.5/Dynare/src/Dynare.jl, in expression starting on line 7
```

**Comment by sebastien-villemot** [comment]
cc @stepan-a : are you interested in maintaining that package?

**Comment by bdemeshev** [comment]
Maybe in the future :) I have just started with dsge models!

**Comment by stepan-a** [comment]
@sebastien-villemot Yes I can do it, if you don't mind that we massively change the current state of the package. We've made some progress in the preprocessor and I would like to use its output here or to compute the derivatives with symengine, a C++ version of SymPy, instead of Calculus.

**Comment by stepan-a** [comment]
I will try to fix the 0.5 issues next week.

**Comment by sebastien-villemot** [comment]
@stepan-a Feel free to make the changes that you think are relevant. However note that this package is registered in the official list of Julia packages, so my feeling is that it should not rely on the Dynare preprocessor for its correct functioning (though it could optionally use it).

**Comment by stepan-a** [comment]
@bdemeshev I think I fixed the compatibility issues with 0.5.x.
